src/browser/provider/built-in/dedicated/chrome/browser-client.ts

```diff
@@ -55,7 +66,19 @@ export class BrowserClient {
 
         this._clients[this._clientKey] = client;
 
+        const timeElapsedStart = process.hrtime();
```

Comment:
We can create an utility function to guard method execution time.
```js

   executionTimeGuard(fn, timeLimit, inspectedMethodName) {
        const timeElapsedStart = process.hrtime();

         await fn();
        
         const timeElapsedFinish = process.hrtime(timeElapsedStart);

         this.debugLogger(`${inspectedMethodName} took ${prettyTime(timeElapsedFinish)}`);

        const [ elapsedSeconds ] = timeElapsedFinish;

        if (elapsedSeconds > timeLimit)
            this._runtimeInfo.providerMethods.reportWarning(WARNING_MESSAGE.browserProviderDropOfPerformance, this._runtimeInfo.browserName);
         
};

```

and use in all appropriate places.

```js
executionTimeGuard(async () => {
     await Page.enable();
}, PAGE_ENABLE_TIME_UPPERBOUND, 'Page.enable');
```